### PR TITLE
Fix IPv6 CIDR based on route

### PR DESCRIPTION
**Pull request by empovit**
Find the "real" CIDR of IPv6 addresses, so that checking if an API VIP
belongs to a subnet on the machine works correctly. 
Otherwise all IPv6 addresses are /128 (one host).

### CODE REVIEW

File: src/apivip_check/apivip_check.go [hunk 1]
```diff
@@ -115,21 +127,43 @@ func getIPByURI(uri string) (net.IP, error) {
...
+
+		if !isVIPv4 {
+			util.SetV6PrefixesForAddress(intf.Name, &routeFinder{}, log, addrStrs)
+		}
+
+		var ipNet *net.IPNet
+		var err error
+		for _, ipAddr := range addrStrs {
+
+			if isVIPv4 {
```
Comment (ori-amizur) [inline src/apivip_check/apivip_check.go:154]: Why is this condition needed? Why not just parse and check the IP is contained in the CIDR?

File: src/inventory/interfaces.go [hunk 2]
```diff
@@ -184,3 +185,15 @@ func (i *interfaces) getInterfaces() []*models.Interface {
 func GetInterfaces(depenndecies IDependencies) []*models.Interface {
 	return newInterfaces(depenndecies).getInterfaces()
 }
+
+func setV6PrefixesForAddresses(interfaces []*models.Interface, dependencies IDependencies) {
+
+	for _, intf := range interfaces {
+
+		if len(intf.IPV6Addresses) == 0 {
```
Comment (ori-amizur) [inline src/inventory/interfaces.go:193]: This condition is not needed

File: src/util/network.go [hunk 3]
```diff
@@ -3,9 +3,121 @@ package util
...
+		return err
+	}
+
+	for i, addr := range addresses {
+
+		if addr == "" {
+			continue
+		}
+
+		ip, _, err := ToIPv6CIDR(addr)
```
Comment (ori-amizur) [inline src/util/network.go:65]: Why was this added?

File: src/inventory/interfaces.go (hunk 2 above)
Comment (empovit) [reply src/inventory/interfaces.go:193]: it was there, probably some kind of optimization

File: src/apivip_check/apivip_check.go (hunk 1 above)
Comment (empovit) [reply src/apivip_check/apivip_check.go:154]: It is needed to skip trying IPv6 subnets with IPv4 API VIP and vice versa. Notice `ipNet != nil` in line 165.

File: src/util/network.go (hunk 3 above)
Comment (empovit) [reply src/util/network.go:65]: I don't want to traverse all routes trying to find the correct CIDR when the IP is not even an IPv6. Line 71.

File: src/inventory/interfaces.go (hunk 2 above)
Comment (empovit) [reply src/inventory/interfaces.go:193]: I believe it's there to avoid fetching a link by name and collecting usable routes if there are no IP addresses to update

**Comment by YuviGold** [comment]
IPv6 isn't being tested from here. I'll add IPv6 job as non-blocking.

File: src/apivip_check/apivip_check.go [hunk 4]
```diff
@@ -7,15 +7,27 @@ import (
...
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
+	"github.com/vishvananda/netlink"
 )
 
 const (
 	WorkerIgnitionPath = "/config/worker"
 )
 
+type routeFinder struct{}
```
Comment (ori-amizur) [inline src/apivip_check/apivip_check.go:21]: It is strange that the logic moved to util but we call netlink here.  Should be in util as well

Comment (empovit) [reply src/apivip_check/apivip_check.go]: moved RouteFinder based on netlink into utils.
